common/compositor/compositordefs.h

```diff
@@ -19,7 +19,10 @@
 
 #include <stdint.h>
 
-#ifdef USE_GL
+#ifdef USE_DC
+#include <EGL/egl.h>
```

Comment:
Why do you need these ?

Reply:
Many errors occur with the EGLImage and Gpu Resource Handle if these aren't added for the struct EGLDisplay and GpuDisplay. Compositorthreads, compositor, and nativesurface try to access these struct values regardless of using opengl, vulkan, or not and don't both checking if the struct has these attributes available. They try to use them and this results in errors in multiple places. I added them initially to quite these errors since files require them regardless of what is being used.

Reply:
If you address comments below and change things to void* you will not need this.